src/realm/obj.cpp

```diff
@@ -2019,9 +2087,28 @@ CollectionPtr Obj::get_collection_ptr(const Path& path) const
     REALM_ASSERT(path.size() > 0);
     // First element in path must be column name
     auto col_key = path[0].is_col_key() ? path[0].get_col_key() : m_table->get_column_key(path[0].get_key());
-    REALM_ASSERT(col_key);
+
+    CollectionBasePtr collection;
     size_t level = 1;
-    CollectionBasePtr collection = get_collection_ptr(col_key);
+    if (col_key) {
```

Comment:
We should probably have an alternative method that takes 'StringData' as an input, otherwise now we need to build a path before calling this or `get_dictionary/list_ptr`

Reply:
I added such a function